linker/linker_test.go

```diff
@@ -2233,6 +2233,12 @@ func TestLinkerValidation(t *testing.T) {
 			},
 			expectedErr: `test.proto:4:5: feature "enum_type" is allowed on [enum,file], not on field`,
 		},
+		"failure_proto3_enum_zero_value": {
```

Comment:
I see that feature targets are already validated. But I noticed that the compiler does not raise an error when the feature `repeated_field_encoding = PACKED` is set on a field that cannot be packed.

Here's a failing test case:

```go
		"failure_edition_repeated_field_encoding_packed_string": {
			input: map[string]string{
				"test.proto": `
					edition = "2023";
					message Foo {
					  repeated string s = 1 [features.repeated_field_encoding = PACKED];
					}
				`,
			},
			expectedErr: `test.proto:3:12: feature PACKED repeated field encoding is only allowed on numeric, boolean, and enum fields`,
		},
```

protoc does raise an error for this case. I haven't gone through all features, but a quick check shows that `utf8_validation` and `message_encoding` have similar constraints. 

The constraints make sense to me, and I believe we should implement them too. I don't see any need to do it in this PR though.

Reply:
Yes, I know this is not implemented yet. (See remark in PR description for  #191, which was the last PR to touch editions-related stuff).

But that's not what this branch is for. That's coming next.